pkg/ingester/metadata_purger.go

```diff
@@ -0,0 +1,7 @@
+package ingester
+
+func (i *Ingester) purgeUsersMetadata() {
+	for _, state := range i.userStates.cp() {
+		state.purgeMetadata(i.cfg.MetadataRetainPeriod)
```

Comment:
The purge process went through my decision-making similarly to the map. There's no priority queue and no concurrency per user. This is deliberate to get it into a reviewable state ASAP - On this one, I'm pretty sure we at least need the per-user concurrency but I'll let you tell me that instead of just assuming it, given I don't have data to work with.